src/rgw/rgw_rest_s3.h

```diff
@@ -119,6 +119,18 @@ class RGWListBucket_ObjStore_S3 : public RGWListBucket_ObjStore {
   void send_versioned_response();
 };
 
+class RGWListBucket_ObjStore_S3v2 : public RGWListBucket_ObjStore {
```

Comment:
i don't see anything that looks for a `?list-type=2` parameter to see whether a request wants list-v2

i also don't see anything that creates one of these RGWListBucket_ObjStore_S3v2 objects

the v1 `RGWListBucket_ObjStore_S3` op gets created by `RGWHandler_REST_Bucket_S3::get_obj_op()` when a GET request is issued against a bucket - so i think that's where we want to check for `list-type=2` and create a `RGWListBucket_ObjStore_S3v2` instead

Reply:
If the marker is empty, I am assigning startAfter to it as it is the first request. Otherwise, I am assigning ContinuationToken to it and nextMarker same as nextContinuationToken. That's why I am inheriting parts of listobjectsv1. Yeah, I need to check for list-type=2. Thanks, Casey